src/tyr/isochrone_serializer.cc

```diff
@@ -85,17 +85,29 @@ serializeIsochrones(const Api& request,
       }));
     }
   }
-  // Add original locations to the geojson
+  // Add input and snapped locations to the geojson
   if (show_locations) {
+    auto snapped_coords = array({});
+    auto input_coords = array({});
     for (const auto& location : request.options().locations()) {
-      features->emplace_back(
-          map({{"type", std::string("Feature")},
-               {"properties", map({})},
-               {"geometry", map({{"type", std::string("Point")},
-                                 {"coordinates", array({fp_t{location.ll().lng(), 6},
-                                                        fp_t{location.ll().lat(), 6}})}})}}));
+      const valhalla::LatLng& latLng_snapped = location.path_edges(0).ll();
+      snapped_coords->push_back(
+          array({fp_t{latLng_snapped.lng(), 6}, fp_t{latLng_snapped.lat(), 6}}));
+
+      const valhalla::LatLng& latlng_input = location.ll();
+      input_coords->push_back(array({fp_t{latlng_input.lng(), 6}, fp_t{latlng_input.lat(), 6}}));
     }
+    features->emplace_back(map(
+        {{"type", std::string("Feature")},
+         {"properties", map({{"type", std::string("input")}})},
+         {"geometry", map({{"type", std::string("MultiPoint")}, {"coordinates", input_coords}})}}));
+
+    features->emplace_back(map(
+        {{"type", std::string("Feature")},
+         {"properties", map({{"type", std::string("snapped")}})},
```

Comment:
if each point was an individual feature we could also specify the index of the location the input and snap points belonged to. it would make the response more verbose but, ultimately more useful.

Reply:
Oops, just reading this now. Actually that's true, that way we could correlate snapped to input points! Good stuff.

I'll do this as soon as I got Valhalla set up again. Had to wipe my Ubuntu a few days ago and went with Arch now, will have to dive more into prime-server deps, didn't work on first try. Should be done after the weekend.